tests/dashbio_demos/app_volcano_plot.py

```diff
@@ -257,13 +343,21 @@ def update_upper_left_number(fig, l_lim):
         return str(number)
 
     @app.callback(
-        Output('vp-upper-right', 'children'),
+        Output('vp-upper-bound-val', 'value'),
+        [Input('vp-upper-bound', 'value')],
+    )
+    def update_upper_bound_val(u_lim):
+        """For selenium tests purpose."""
+        return u_lim
+
+    @app.callback(
+        Output('vp-upper-right-val', 'children'),
         [Input('vp-graph', 'figure')],
         [State('vp-upper-bound', 'value')]
     )
-    def update_upper_right_number(fig, u_lim,):
+    def update_upper_right_number_val(fig, u_lim,):
         """Update the number of points in the upper right zone delimited by
-        the thresholds.
+        the thresholds for tests purpose.
```

Comment:
I'm not so sure it's a good idea to have callbacks/app features specifically so that they will work with the tests. In my opinion, it's far more important for the components to be continuously tested. Since these apps (and their code) will be on display for the public, it would be better to remove these (and, if necessary, the corresponding tests from the testing file). Otherwise, it might look confusing as these callbacks have no purpose outside of testing.

Reply:
I very much agree with @shammamah here.

Reply:
I didn't know the code of the demo would be shared publicly, my idea was that it would be good to guarantee that the app is not broken using tests so that we don't have to go every time through all the apps, testing them manually to make sure they are fine. I will simply reset the head to not include the tests then :)

Reply:
Yeah, sorry about not mentioning that! If you go to the apps on the Dash Bio portal (which is public), there is a little "View on GitHub" link at the top right. It's similar to the Dash DAQ demo apps that we created -- people should ideally be able to clone the app repo and just run the app on their own with the ability to understand what each portion of the code is doing. :)

Reply:
@Bachibouzouk I think your intention was good: If I'm not mistaken, your intention here was to program defensively, by using an ad-hoc callback in the same spirit that we include assertions, right? The idea is good, but I would see these 'tests' either as more native to the demo app, or completely separated (in a different, test-only file).

Reply:
Yes that was my intention. If the test are completely separated then if someone changes the app, the tests need to be changed again. I think we could rather use similar [functions](https://github.com/plotly/dash-cytoscape/blob/master/tests/test_interactions.py) than those @xhlulu implemented to test cytoscape to change the values of sliders. The only problem remaining is that the `daq.LEDDisplay` css doesn't contain an explicit value of the number it represents, so one would still need a hidden Div or a Store solely for test purpose.

Reply:
@Bachibouzouk Yes, but that's exactly the point of a test suite; you don't want changes that are not backward-compatible to go unnoticed... So, if they force you to update the tests, that's good, that's what we want.

Reply:
So, I haven't wrapped my head around those tests; for example, it seems frustrating to convert a number into a string and then back into an integer, but I guess it's the only way to store and pass the value... @shammamah I suggest leaving this PR as is, so at least we benefit from @Bachibouzouk's improvements asap; then I guess you'll want to apply the 'standard' redesign to this demo app; and we can investigate these integration tests at some point in the future.

Reply:
@mkcor Sounds good to me!